NGitLab/Models/UserQuery.cs

```diff
@@ -46,5 +49,56 @@ public class UserQuery
         /// (Admin only) Order users in asc or desc order. Default is descending
         /// </summary>
         public string Sort { get; set; }
+
+        /// <summary>
+        /// (Admin only) Search users based on external UID
+        /// </summary>
+        public string ExternalUid { get; set; }
+
+        /// <summary>
+        /// (Admin only) Search users based on Provider
+        /// </summary>
+        public string Provider { get; set; }
+
+        /// <summary>
+        /// (Admin only) Search for users without projects
+        /// </summary>
+        public bool? WithoutProjects { get; set; }
+
+        /// <summary>
+        /// (Admin only) Search for users created before specified date time
+        /// Format: "YYYY-MM-DDThh:mm:ss.sTZD" eg. "2001-01-02T00:00:00.060Z"
+        /// </summary>
+        public DateTime? CreatedBefore { get; set; }
+
+        /// <summary>
+        /// (Admin only) Search for users created after specified date time
+        /// Format: "YYYY-MM-DDThh:mm:ss.sTZD" eg. "2001-01-02T00:00:00.060Z"
+        /// </summary>
+        public DateTime? CreatedAfter { get; set; }
+
+        /// <summary>
+        /// (Admin only) Include the users custom attributes in the response
+        /// </summary>
+        public bool? WithCustomAttributes { get; set; }
+
+        /// <summary>
+        /// (Admin only) Gets users that have two factor either enabled or disabled
+        /// Possible values: "enabled", "disabled"
+        /// </summary>
+        public TwoFactorState? TwoFactor { get; set; }
+
+        /// <summary>
+        /// (Admin only)If true, get only users that are Admin
+        /// </summary>
+        public bool? IsAdmin { get; set; }
+
+        public enum TwoFactorState
```

Comment:
You should move the enum out of the `User` type. Nested types are often harder to use/discover.